src/chemistry/network/kida.hpp

```diff
@@ -0,0 +1,226 @@
+#ifndef CHEMISTRY_NETWORK_KIDA_HPP_
+#define CHEMISTRY_NETWORK_KIDA_HPP_
+//========================================================================================
+// Athena++ astrophysical MHD code
+// Copyright(C) 2014 James M. Stone <jmstone@princeton.edu> and other code contributors
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+//! \file H2.hpp
+//! \brief definitions for a general chemical network in KIDA format
+
+//c++ headers
+#include <map>    //map
+#include <string> //std::string
+#include <vector> // vector container
+
+// Athena++ classes headers
+#include "../../athena.hpp"
+#include "../../athena_arrays.hpp"
+#include "../utils/kida_reaction.hpp"
+#include "../utils/kida_species.hpp"
+#include "network.hpp"
+
+class Units;
+
+//reaction types
+enum class ReactionType {none, cr, crp, photo, twobody, twobodytr,
+                         grain_implicit, special, grain_collision};
+
+//! \class ChemNetwork
+//! \brief General chemical network in KIDA format
+class ChemNetwork : public NetworkWrapper {
+  //It would be convenient to know the species names in
+  //initialization of chemical species in problem
+  friend class MeshBlock;
+ public:
+  ChemNetwork(MeshBlock *pmb, ParameterInput *pin);
+  ~ChemNetwork();
+
+  //a list of species name, used in output
+  std::string species_names[NSPECIES];
```

Comment:
should this be `static const` like it is in gow17, G14sod, H2?

Reply:
No, this is evaluated at runtime by reading in a text file. In gow17, G14Sod, H2, the species names are initialized with constant values in .cpp files.

Reply:
gotcha, maybe an explicit comment for developers that it is unique?